### PR TITLE
Add code review guidelines

### PR DESCRIPTION
**Pull request by robertoaloi**
### Description

Since the amount of collaborators is growing, I believe is good to write down some guidelines for reviewers, too. These are simply my own suggestions, so feel free to argue against them, if you believe they are not what we want.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -101,6 +101,52 @@ A good commit may look like this:
...
+  block_. The Pull Request author is encouraged to fix nit-picking
+  comments, but they should not be considered as blockers.
+* It is not acceptable to mark a Pull Request as a _Request Changes_
+  in case only nit-picking comments are present.
+* In case of post-merge comments, the original author of the Pull
+  Request must be pinged directly via a `@` mention.
+* It is acceptable to open a Pull Request to get some early feedback
+  from other team members, even if the changes included in the Pull
+  Request are not in a _reviewable_ state, yet. In this case, though,
+  the Pull Request title _must_ be prefixed with `[WIP]`.
```
Comment (alanz) [inline .github/CONTRIBUTING.md]: What about github draft pull requests?

Comment (robertoaloi) [reply .github/CONTRIBUTING.md]: I didn't know they existed O_o

https://github.blog/2019-02-14-introducing-draft-pull-requests/

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -101,6 +101,52 @@ A good commit may look like this:
 
     [#42] Discover the meaning of life.
 
+### Code Reviews
+
+* Always be constructive when providing feedback (e.g. be explicit on
+  what could improved and how)
```
Comment (michalmuskala) [inline .github/CONTRIBUTING.md]:
```suggestion
  what could be improved and how)
```

**Comment by michalmuskala** [comment]
I think there needs to be some space for fly-by reviews, especially around typos or other minor issues like that from people that can't directly review (since they don't have commit bits), or don't want to blanket accept because they don't feel they know enough of the particular area to fully accept a PR. Alternatively this should state clearly that such reviews shouldn't be done.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -101,6 +101,52 @@ A good commit may look like this:
...
+* Use [GitHub suggestion blocks][github-suggest] for fixing typos and
+  small mistakes
+* Positive feedback is as important as negative feedback
+* Only the Pull Request author(s) should push changes to the branch
+  under review
+* Reviewers should be as responsive as possible in providing their
+  reviews.
+* Every review should be completed via either an _Approve_ or a
```
Comment (jfacorro) [inline .github/CONTRIBUTING.md]: What about "Comment"?

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -101,6 +101,52 @@ A good commit may look like this:
...
+  _Request Changes_ action. Not selecting either option should be
+  considered as a _review in progress_.
+* Code reviews should be completed within a reasonable time
+  window. Leaving comments in time-separated _bulks_ is discouraged.
+* It is not acceptable to mark a Pull Request via a _needs work_
+  without further explanation.
+* PR Conversations can be marked as _resolved_ by the PR author, but
+  only if at least one of the following conditions holds:
+    * The issue is addressed via a code change;
+    * the issue will be tackled in a follow up Pull Request (a
```
Comment (jfacorro) [inline .github/CONTRIBUTING.md]:
```suggestion
    * The issue will be tackled in a follow up Pull Request (a
```

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -101,6 +101,52 @@ A good commit may look like this:
...
 
+### Code Reviews
+
+* Always be constructive when providing feedback (e.g. be explicit on
+  what could improved and how)
+* Use [GitHub suggestion blocks][github-suggest] for fixing typos and
+  small mistakes
+* Positive feedback is as important as negative feedback
+* Only the Pull Request author(s) should push changes to the branch
+  under review
```
Comment (alanz) [inline .github/CONTRIBUTING.md]: Do we want to say anything about a rebase policy?

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (robertoaloi) [reply .github/CONTRIBUTING.md:125]: Going for the other way around. After a `;` using a lowercase.

File: .github/CONTRIBUTING.md (hunk 5 above)
Comment (robertoaloi) [reply .github/CONTRIBUTING.md:112]: GitHub seems to handle most of the cases by itself. We can add something, yes. Maybe as a follow-up?

**Comment by robertoaloi** [comment]
> I think there needs to be some space for fly-by reviews, especially around typos or other minor issues like that from people that can't directly review (since they don't have commit bits), or don't want to blanket accept because they don't feel they know enough of the particular area to fully accept a PR. Alternatively this should state clearly that such reviews shouldn't be done (which to me would also be fine).

Clarified that this applies to project contributors and/or reviewers whose review has been explicitly requested.

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (robertoaloi) [reply .github/CONTRIBUTING.md:115]: Mentioned.

**Pull request merged**